x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/queries/TextExpansionQueryBuilder.java

```diff
@@ -227,14 +222,24 @@ protected QueryBuilder doRewrite(QueryRewriteContext queryRewriteContext) throws
     }
 
     private QueryBuilder weightedTokensToQuery(String fieldName, TextExpansionResults textExpansionResults) {
-        WeightedTokensQueryBuilder weightedTokensQueryBuilder = new WeightedTokensQueryBuilder(
-            fieldName,
-            textExpansionResults.getWeightedTokens(),
-            tokenPruningConfig
-        );
-        weightedTokensQueryBuilder.queryName(queryName);
-        weightedTokensQueryBuilder.boost(boost);
-        return weightedTokensQueryBuilder;
+        if (tokenPruningConfig != null) {
```

Comment:
Could you add a comment that mentions the CCR or mixed version cluster use case - I believe that's the only case when `tokenPruningConfig` might be null?

Reply:
No, token pruning config being null is the default, but it turns out that CCS breaks since it supports 8.12 and token pruning/weighted tokens was introduced in 8.13.